# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [5]:
rows = data['total_income'] == '\n'
print(data.loc[rows])

Empty DataFrame
Columns: [children, days_employed, dob_years, education, education_id, family_status, family_status_id, gender, income_type, debt, total_income, purpose]
Index: []


In [6]:
data['total_income'].unique

<bound method Series.unique of 0        253875.639453
1        112080.014102
2        145885.952297
3        267628.550329
4        158616.077870
             ...      
21520    224791.862382
21521    155999.806512
21522     89672.561153
21523    244093.050500
21524     82047.418899
Name: total_income, Length: 21525, dtype: float64>

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [7]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [8]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [9]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [10]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [11]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [12]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [13]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [14]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [15]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [16]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [17]:
data.duplicated().sum()

71

In [18]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [19]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [20]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [21]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [22]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [23]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [ ]:
tmp = data.groupby('children')['debt'].agg(['sum','count'])

tmp 

In [27]:
tmp['ratio'] = tmp['sum'] / tmp['count']

tmp

sum  count     ratio
children                       
0         1063  14091  0.075438
1          444   4808  0.092346
2          194   2052  0.094542
3           27    330  0.081818
4            4     41  0.097561
5            0      9  0.000000

**Вывод:** В ходе проверки гипотезы о зависимости количества детей и появлением задолженности можно ошибочно сделать вывод,что люди имеющие четверых детей чаще всего становятся должниками и их почти 10% от общего количества, но такой вывод будет преждевременным, так как выборка на которой производится исследование слишком маленькая и требуется больше данных для окнчательного корректного вывода. На что стоит обратить внимания в данный момент исходя из предоставленных данных и проведенного анализа: люди не имеющие детей,которых в общей сложности 14091 человек имеют просрочки в 7.54% случаев, люди имеющие одного ребенка,которых 4808 человек,имеют просрочки в 9.23% случаев, люди имеющие двух детей,которых в данной выборке 2052 человека,имеют просрочки в 9.45% случаев, то есть взяв эти три группы людей,можно сделать вывод,что в их случаи с появлением большего количества детей процент просрочки растет. Группа,которая включает в себя людей имеющих 3 детей, имеет один из наиболее маленьких процентов по количеству должников: 8.18%, возможно это связано с тем,что люди с тремя детьми начинают более тщательно планировать свои финансы и являются более финансово грамотными или же данных также недостаточно,как и в случаи с группами с 4 и 5 детьми. Вывод о реальных процентах должников в группах людей с четырьмя и пятью детьми на данный момент,как я уже сказал выше, невозможно сделать, так как требуется больше данных. Сделать окончательный вывод о зависимости количества детей и возвратом кредита в срок на данный момент будет неверно,нужно больше данных.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [36]:
print(data.groupby('family_status_id')['debt'].agg(['sum','count']))

print(f'{927/12261:.2%} - доля должников в браке')
print(f'{385/4134:.2%} - доля должников в гражданском браке')
print(f'{63/951:.2%} - доля должников,которые являются вдовами \ вдовцами')
print(f'{84/1189:.2%} - доля должников в разводе')
print(f'{273/2796:.2%} - доля должников, не женатых\ не замужем')

                  sum  count
family_status_id            
0                 927  12261
1                 385   4134
2                  63    951
3                  84   1189
4                 273   2796
7.56% - доля должников в браке
9.31% - доля должников в гражданском браке
6.62% - доля должников,которые являются вдовами \ вдовцами
7.06% - доля должников в разводе
9.76% - доля должников, не женатых\ не замужем


**Вывод:**  Исходя из цифр,которы мы получили в ходе анализа,можно сделать вывод: существует две группы самых злостных должников - люди которые официально никак не закрепили свои отноешения,а именно люди в гражданском браке ( процент должников от общего количества 9.31%) и люди не женаты / не замужен (процент должников от общего количества 9.76%),то есть люди этих двух групп в большей степени являются должниками. Оставщиеся группы людей имеют такие показатели: люди в браке 7.56% должников от общего количества, люди,которые являются вдовами\вдовцами 6.62% от общего количества(что кстати, самый низкий показатель среди всех групп,но и группа самая малочисленная), люди в разводе 7.06% от общего количества. То есть люди,которые официально связывали свои отношения реже становятся должниками.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [30]:
total_income_debt_A = data[(data['total_income_category'] == 'A') & (data['debt'] == 1)]['debt'].count() # подсчет количества людей согласно каждой категории,имеющих задолженности по кредитам
total_income_debt_B = data[(data['total_income_category'] == 'B') & (data['debt'] == 1)]['debt'].count()
total_income_debt_C = data[(data['total_income_category'] == 'C') & (data['debt'] == 1)]['debt'].count()
total_income_debt_D = data[(data['total_income_category'] == 'D') & (data['debt'] == 1)]['debt'].count()
total_income_debt_E = data[(data['total_income_category'] == 'E') & (data['debt'] == 1)]['debt'].count()

total_category_A = data[data['total_income_category'] == 'A'].value_counts().count() # подсчет общего количества людей в каждой категории
total_category_B = data[data['total_income_category'] == 'B'].value_counts().count()
total_category_C = data[data['total_income_category'] == 'C'].value_counts().count()
total_category_D = data[data['total_income_category'] == 'D'].value_counts().count()
total_category_E = data[data['total_income_category'] == 'E'].value_counts().count()

print(f'Процент людей имеющих уровень дохода A и имеющих задолженоость по кредиту = {total_income_debt_A/total_category_A:.2%}') # Соотнощению в процентах общего количества людей по категориям к общему количеству людей в этих категориях, имеющих задолженности по кредитам
print(f'Процент людей имеющих уровень дохода B и имеющих задолженоость по кредиту = {total_income_debt_B/total_category_B:.2%}')
print(f'Процент людей имеющих уровень дохода C и имеющих задолженоость по кредиту = {total_income_debt_C/total_category_C:.2%}')
print(f'Процент людей имеющих уровень дохода D и имеющих задолженоость по кредиту = {total_income_debt_D/total_category_D:.2%}')
print(f'Процент людей имеющих уровень дохода E и имеющих задолженоость по кредиту = {total_income_debt_E/total_category_E:.2%}')


Процент людей имеющих уровень дохода A и имеющих задолженоость по кредиту = 8.00%
Процент людей имеющих уровень дохода B и имеющих задолженоость по кредиту = 7.06%
Процент людей имеющих уровень дохода C и имеющих задолженоость по кредиту = 8.50%
Процент людей имеющих уровень дохода D и имеющих задолженоость по кредиту = 6.02%
Процент людей имеющих уровень дохода E и имеющих задолженоость по кредиту = 9.09%


**Вывод:** Люди чьи доходы относятся к категории E(30000-50000), чаще чем остальные являются нарушителями сроков выплаты кредитов, когда в свою очередь люди, чиь доходы относятся к категории D(50000-200000) реже всего имеют просрочки по кредитам, то есть можно утверждать,что категория доходов влияет на вероятность выплаты кредита в назначенный срок.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
purpose_build_debt_count = data[(data['purpose_category'] == 'операции с недвижимостью') &  
                                    (data['debt'] == 1)]['debt'].count() # подсчет людей по категориям цели кредита,которые не выплатили его вовремя

purpose_avto_debt_count = data[(data['purpose_category'] == 'операции с автомобилем') & 
                                    (data['debt'] == 1)]['debt'].count()

purpose_education_debt_count = data[(data['purpose_category'] == 'получение образования') & 
                                    (data['debt'] == 1)]['debt'].count()

purpose_wedding_debt_count = data[(data['purpose_category'] == 'проведение свадьбы') & 
                                    (data['debt'] == 1)]['debt'].count()


purpose_build_total_count = data[data['purpose_category'] == 'операции с недвижимостью'].value_counts().count() # подсчет людей по категориям цели кредита

purpose_avto_total_count = data[data['purpose_category'] == 'операции с автомобилем'].value_counts().count() 

purpose_education_total_count = data[data['purpose_category'] == 'получение образования'].value_counts().count()

purpose_wedding_total_count = data[data['purpose_category'] == 'проведение свадьбы'].value_counts().count() 


print(f'Процент людей, которые берут кредиты на недвижимость и не возвращают в срок = {purpose_build_debt_count/purpose_build_total_count:.2%}')
print(f'Процент людей, которые берут кредиты на автомобиль и не возвращают в срок = {purpose_avto_debt_count/purpose_avto_total_count:.2%}')
print(f'Процент людей, которые берут кредиты на образование и не возвращают в срок = {purpose_education_debt_count/purpose_education_total_count:.2%}')
print(f'Процент людей, которые берут кредиты на свадьбу и не возвращают в срок = {purpose_wedding_debt_count/purpose_wedding_total_count:.2%}')


Процент людей, которые берут кредиты на недвижимость и не возвращают в срок = 7.26%
Процент людей, которые берут кредиты на автомобиль и не возвращают в срок = 9.35%
Процент людей, которые берут кредиты на образование и не возвращают в срок = 9.25%
Процент людей, которые берут кредиты на свадьбу и не возвращают в срок = 7.91%


**Вывод:** Просчитав зависимость по каждой из групп людей поделенных по цели кредита,можно сделать вывод,что наиболее часто не выплачивают кредит в срок люди, чья цель была покупка автомобиля, доля должников из общего количества людей в этой группе составялет 9.35%, а это в данной выборке 400 человек из 4279,почти каждый 10 человек, что на 2% больше,чем в группе, где целью кредита является покупка недвижемости 7.26%,далее по нарастающей идет группа людей,где целью кредита является организация свадьбы 7.91% и ближе всего в автолюбителям идет группа людей,которая берет займ для получения образования - 9.25%. Хоть по итогу анализа, люди берущие кредиты на авто и на образования чаще не возвращают их в срок,разница в два процента от большего и меньшего значения на данном этапе исселодования не гарантированно нам указвает на платежеспособность каждой группы.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

- 1. Технологические ошибки возникают при копировании, считывании, смене формата.
- 2. Человеческий фатор - при самостоятельном вводе данных любой человек рано или поздно ошибается
- 3. При заполнении анект в социальных опросах, некоторые столбцы, например, могут специально не заполнять из-за ненадобности, при этом все остальные будут заполнены.
- 4. Некорректная преддоработка данных при анализе
         

In [32]:
data['income_type'].unique()

array(['сотрудник', 'пенсионер', 'компаньон', 'госслужащий',
       'безработный', 'предприниматель', 'студент', 'в декрете'],
      dtype=object)

In [33]:
data['days_employed'] = data['days_employed'].fillna('unkdown')

In [34]:
data['total_income'] = data['total_income'].fillna('unkdown')

In [35]:
data[(data['days_employed'] == 'unkdown')  & (data['total_income'] == 'unkdown') & (data['education_id'] == 1)]

Empty DataFrame
Columns: [children, days_employed, dob_years, education, education_id, family_status, family_status_id, gender, income_type, debt, total_income, purpose, total_income_category, purpose_category]
Index: []

**Вывод:** Cогласно высказанным мной гипотезам о гарантированно существующей зависимости между,например, уровнем дохода и сроком возврата кредита в срок 100% зависимости не существует и пропуски,которые есть в исходных данных,связаны с тем,что заполнение данных по определенным столбцам было необязательно при оформлении заявки,так как эти данные либо не берутся в расчет, либо берутся из других более достоверных источников. То есть, я полагаю,что пропуски в предоставленных нам исходных данных обусловленны тем, что при не заполнении столбцов 'total_income' и 'days_employed' заявка все равно принимается к рассмотрению. Что еще мне показалось любопытным,так это то,что большенсто людей,которые не заполнили эти столбцы, а это 1500 из 2000 почти 75%, указали уровень образования 'Средний', не уверен,что здесь есть какая-то гарантированная взаимосвязь.  Нужно больше данных

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медианные значения при заполнении пропусков в преддоработке данных , в сравнении, например, с средними арифметическими использовать корректнее из-за того,что медаинные значения не так подверженны воздействию аномальных выбросов, то есть при замене пропусков на медианные, никакие аномалии не послужат в дальнейшем не верным выводам. Арифметические значения в свою очередь сильно подвержены изменениям из-за аномальных выборосов и заполнение ими пропуском в дальнейшем повредят к не корректным выводам. 

### Шаг 4: общий вывод.


По предоставленным исходным данным, был проведен анализ с целью разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов. В дальнейшем результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

Было выдвинуто 4 гипотезы:

- Гипотеза первая : влияет ли количество детей на возврат кредита в срок
- Гипотеза вторая : влияет ли семейное положение на возврат кредита в срок
- Гипотеза третья : влияет ли уровень дохода на возврат кредита в срок
- Гипотеза четвертая : влияют ли цели кредита на возврат кредита в срок

В ходе исследования удалось выявить следущее:

<b>Гипотеза первая частично подтверждена:</b>


Окончательно подтвердить первую гипотезу или опровергуть я смогу если получу больше данных по группам людей с 3, 4 и 5 детьми, на данном этапе исследования результаты такие:
- 7.54% - доля просроченных кредитов у заёмщиков без детей
- 9.23% - доля просроченных кредитов у заёмщиков с одним ребенком
- 9.45% - доля просроченных кредитов у заёмщиков с двумя детьми 

То есть прослеживается тенденция по увеличению процента должников с увеличением количества детей, вместе с тем,присутсвующая разница в 2% не даёт полной уверенности в подверждении гипотезы из-за отсутствия полноты данных по следующим группам.
Рекомендации: полагаться на этот показатель только в случаи совокупного принятия решения. Предоставить больше данных для анализа. 

<b>Гипотеза вторая подтверждена:</b>


- Люди никогда не состаявщие в законном браке,а именно не женатые / не замужем и проживающие в гражданском браке чаще становятся должниками по своим кредитным обязательствам. Их соответственно 9.76% от общего количества и 9.31% от общего количества, то есть практически каждый 10 человек.
- Люди,чьи отнощения находятся в официальном браке , являются вдовами / вдовцами или находятся в разводе, являются более надежными заёмщиками, не платят кредиты в 7.56%, 6.62% и 7.06% случаях соответственно. 

Рекомендации: Разница от меньшего значения до большего составляет 3%, этот показатель стоит взять во внимание,но принимать решение исключительно по данному критерию было бы некорректно. 

<b>Гипотеза третья подтверждена частично:</b>


- Люди, чьи доходы относятся к категории E(30000-50000), чаще чем остальные являются нарушителями сроков выплаты кредитов, в 9.09% случаев.
- Люди, чьи доходы относятся к категории D(50000-200000) реже всего имеют просрочки по кредитам, в 6.02% случаев.

Рекомендации: Тот факт,что люди с наименьшим количеством доходов ,категория E(30т.-50т.), имеют чаще всего проблемы с выплатами кредита в срок ожидаемо, люди с доходом категории D(50 т. - 200 т.) скорее всего становятся более финансово грамотными и распределяют финансы заранее. Что на самом деле интересно,так это то,что люди имеющие более высокие доходы по сравнению с группой D, скорее всего начинают меньше следить за своими финансами,а тем более за выплатами кредитов в срок,но категории с большими дохода C,B и A малочиселнны, в связи с этим сложно окончательно ответить подтверждена ли окончательно гипотеза о зависимости доходов и выплаты кредита в срок. Мои рекомендации: также не полагаться исключительно на этот фактор при кредитном скоринге, так как вопрос не окончательно изучен.

<b>Гипотеза четвертая подтверждена:</b>

- Процент людей, которые берут кредиты на недвижимость и не возвращают в срок = 7.26%
- Процент людей, которые берут кредиты на автомобиль и не возвращают в срок = 9.35%
- Процент людей, которые берут кредиты на образование и не возвращают в срок = 9.25%
- Процент людей, которые берут кредиты на свадьбу и не возвращают в срок = 7.91%

Рекомендации: Существует зависимость между целью кредита и появлением задолженности, но стоит учитывать также,что разница между наименьшим значением и наибольшем составляет 2%, в связи с этим,также не стал бы рекомендовать полагаться исключительно на этот пунтк при принятии решения.


<b>Общие рекомендации и итоги:</b>

- Проведнные нами иследования, выявиили,что по некоторым гипотезам действительно существует конкретная зависимость,что позволяет нам подтвердить гипотезу вторую и гипотезу четвертую.
- Гипотеза первая и третья могут быть подтверджены только частично из-за неполноты информации в исходных данных.